var/spack/repos/builtin/packages/hip/package.py

```diff
@@ -47,7 +48,7 @@ class Hip(CMakePackage):
     patch('0001-Make-it-possible-to-specify-the-package-folder-of-ro.patch', when='@3.5.0:')
 
     # See https://github.com/ROCm-Developer-Tools/HIP/pull/2141
-    patch('0002-Fix-detection-of-HIP_CLANG_ROOT.patch', when='@3.5.0:')
+    patch('0002-Fix-detection-of-HIP_CLANG_ROOT.patch', when='@:3.9.0')
```

Comment:
It's good that this is fixed now, but why they don't just merge my PR is beyond me.